### PR TITLE
DA: Executor http api

### PR DESCRIPTION
**Pull request by bacv**
## 1. What does this PR implement?

This pr implements a separate executor http api, which reuses common endpoint handlers from the nomos node.

## Checklist

* [x] 1. Description added.
* [x] 2. Context and links to Specification document(s) added.
* [x] 3. Main contact(s) (developers and specification authors) added
* [x] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 5. Link PR to a specific milestone.

### CODE REVIEW

File: nodes/nomos-executor/Cargo.toml [hunk 1]
```diff
@@ -4,17 +4,42 @@ version = "0.1.0"
...
 uuid = { version = "1.10.0", features = ["v4"] }
-serde_yaml = "0.9.34+deprecated"
 
 [features]
 default = ["tracing"]
 mixnet = ["nomos-node/mixnet"]
-metrics = ["nomos-node/metrics"]
-tracing = ["nomos-node/tracing"]
\ No newline at end of file
+metrics = []
```
Comment (danielSanchezQ) [inline nodes/nomos-executor/Cargo.toml:44]: shouldn't we activate the nomos-node lib metrics feature as well? :thinking:

File: nodes/nomos-executor/src/api/backend.rs [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+pub struct AxumBackend<
+    A,
+    B,
+    C,
+    M,
+    V,
+    VB,
+    T,
+    S,
```
Comment (danielSanchezQ) [inline nodes/nomos-executor/src/api/backend.rs:51]: Now I know my ABC :ab:

File: nodes/nomos-node/src/api/backend.rs [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+                        SamplingNetworkAdapter,
+                        SamplingRng,
+                        SamplingStorage,
+                        SIZE,
+                    >,
+                ),
+            )
+            .route("/da/add_blob", routing::post(add_blob::<A, B, M, VB, S>))
+            .route(
+                "/da/get_range",
```
Comment (danielSanchezQ) [inline nodes/nomos-node/src/api/backend.rs:258]: Yeah, nice changes. Better to compose here than the over-abstraction of converting the endopoints in types. It was my fault though, have been thinking about this for time. So, nice catch.

**Review by danielSanchezQ — APPROVED** [review]
Nice :rocket: !

File: nodes/nomos-executor/Cargo.toml (hunk 1 above)
Comment (bacv) [reply nodes/nomos-executor/Cargo.toml]: For completeness I've added the feature back, but basically we are not using `nomos-node/metrics` related code in executor. Each node constructs their own registry according to the feature flag at the moment. I'll revisit the feature flags when working on the metrics expansion.

Comment (danielSanchezQ) [reply nodes/nomos-executor/Cargo.toml]: Cool. I just wanted to understand as I was missing something indeed. Thanks!

**Pull request merged**